cl/recap/tasks.py

```diff
@@ -1979,30 +1961,11 @@ def fetch_pacer_doc_by_rd_base(
         return
 
     # Logic to replicate the PDF sub-dockets matched by RECAPDocument
-    sub_docket_main_rds = list(
-        get_main_rds(court_id, pacer_doc_id).exclude(
-            docket_entry__docket__pacer_case_id=pacer_case_id
-        )
+    subdocket_pqs_to_replicate = find_subdocket_pdf_rds_from_data(
+        fq.user_id, court_id, pacer_doc_id, [pacer_case_id], pdf_bytes
     )
-    sub_docket_pqs = []
-    for main_rd in sub_docket_main_rds:
-        # Create PQs related to RD that require replication.
-        sub_docket_pqs.append(
-            ProcessingQueue(
-                uploader_id=fq.user_id,
-                pacer_doc_id=main_rd.pacer_doc_id,
-                pacer_case_id=main_rd.docket_entry.docket.pacer_case_id,
-                document_number=main_rd.document_number,
-                attachment_number=main_rd.attachment_number,
-                court_id=court_id,
-                upload_type=UPLOAD_TYPE.PDF,
-                filepath_local=ContentFile(pdf_bytes, name="document.pdf"),
-            )
-        )
-
-    if sub_docket_pqs and not is_appellate_court(court_id):
-        pqs_created = ProcessingQueue.objects.bulk_create(sub_docket_pqs)
-        replicate_fq_pdf_to_subdocket_rds.delay([pq.pk for pq in pqs_created])
+    if subdocket_pqs_to_replicate and not is_appellate_court(court_id):
+        replicate_fq_pdf_to_subdocket_rds.delay(subdocket_pqs_to_replicate)
```

Comment:
I'm concerned about this line. If we start creating queues for sub-dockets and then the `replicate_fq_pdf_to_subdocket_rds` task isn't executed due to the `is_appellate_court` helper returning `true`, will we have unprocessed processing queue records?